pkg/certificate/providers/vault/certificate_manager_test.go

```diff
@@ -114,99 +48,115 @@ var _ = Describe("Test client helpers", func() {
-			certBytes := uuid.New().String()
-			issue := func(certificate.CommonName, time.Duration) (*certificate.Certificate, error) {
-				cert := &certificate.Certificate{
-					IssuingCA:    pem.RootCertificate(certBytes),
-					SerialNumber: certSerialNum,
+	testCases := []struct {
+		description string
+		wantErr     bool
+	}{
+		{description: "no err", wantErr: false},
+		{description: "all nil", wantErr: true},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.description, func(t *testing.T) {
+			if tc.wantErr {
+				_, err := New("", "", "")
+				if err == nil {
+					t.Fatalf("expected error, got nil")
+				}
+				_, err = New(addr, token, vaultRole)
+				if err != nil {
+					t.Fatalf("did not expect error, got %v", err)
 				}
-				return cert, nil
 			}
-			issuingCA, serialNumber, err := cm.getIssuingCA(issue)
-			Expect(err).ToNot(HaveOccurred())
-
-			// Ensure that cache is NOT affected
-			Expect(issuingCA).To(Equal([]byte(certBytes)))
-			Expect(len(getCachedCertificateCNs())).To(Equal(expectedNumberOfCertsInCache))
-			Expect(getCachedCertificateCNs()).To(ContainElement(certificate.CommonName("this.has.expired")))
-			Expect(getCachedCertificateCNs()).To(ContainElement(certificate.CommonName("valid.certificate")))
-			Expect(serialNumber).To(Equal(certSerialNum))
 		})
+	}
+}
 
-		It("gets certs from cache", func() {
-			// This cert does not exist - returns nil
-			Expect(cm.getFromCache("nothing")).To(BeNil())
-
-			// This cert has expired -- returns nil
-			Expect(cm.getFromCache(expiredCertCN)).To(BeNil())
+func Test_IssueCertificate(t *testing.T) {
+	var common_name certificate.CommonName = "localhost"
+	var validityPeriod = time.Hour
+	//start vault with pki backend
 
-			actual := cm.getFromCache(validCertCN)
-			Expect(actual).To(Equal(validCert))
+	core, token, addr := mockVault(t)
+	defer core.Shutdown()
+	cm, err := New(addr, token, vaultRole)
+	if err != nil {
+		t.Fatalf("did not expect error, got %v", err)
+	}
+	mockVaultwithPKI(t, cm)
+
+	testCases := []struct {
+		description string
+		wantErr     bool
+	}{
+		{description: "no err", wantErr: false},
+		{description: "all nil", wantErr: true},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.description, func(t *testing.T) {
+			if tc.wantErr {
+				_, err = cm.IssueCertificate("", validityPeriod)
+				if err == nil {
+					t.Fatalf("expected error, got nil")
+				}
+			}
+			_, err = cm.IssueCertificate(common_name, validityPeriod)
+			if err != nil {
+				t.Fatalf("did not expect error, got %v", err)
+			}
 		})
+	}
 
-		It("creates certificates", func() {
-			actual, err := cm.GetCertificate(validCertCN)
-			Expect(err).ToNot(HaveOccurred())
-			Expect(actual).To(Equal(validCert))
-		})
+}
 
-		It("can fetch root cert", func() {
-			actual, err := cm.GetRootCertificate()
-			Expect(err).ToNot(HaveOccurred())
-			Expect(actual).To(Equal(rootCert))
-		})
+func mockVault(t *testing.T) (*vault.Core, string, string) {
```

Comment:
That's Vault running - why do you find them not useful to what we're testing?

Reply:
Any log that's not an error seems to indicate that Vault is running as expected, so that seems like pure noise to me. I imagine if there's an easy way to turn them off it should be simple to turn them on again when debugging these tests locally if there's any doubt Vault isn't working right where those extra logs would be helpful.

Again, not a big deal though if there's no obvious way to adjust the logs.